Description of OOP Modal in code:

scokpius opened the pull request:
- entity classes created
- the attributes and methods necessary to work with them are described
- added getters and setters

vladimir-moshkov commented on Task_Animate_the_OOP_model/src/main/java/Guest.java:
```diff
@@ -1,31 +1,31 @@
-    public boolean isAdultGuest(int age){
-        if (age > 17 ) {return true;};
+    public boolean isAdultGuest(int age) {
+        if (age > 17) {
```
Подумать, как можно упростить всю эту конструкцию

vladimir-moshkov commented on Task_Animate_the_OOP_model/src/main/java/Hotel.java:
```diff
@@ -4,6 +4,7 @@
     private String hotelName;
     private HotelRoom[] listRooms;
     private String[] photoHotel;
```
Фото отеля - это картинка? 
Почему массив стрингов?

scokpius replied:
Во время выполнения задачи по созданию диаграммы классов было оговорено, что  фото это не картинка а ссылка на нее и в диаграмме она идет как ImageURL, так как том не одна картинка то следовательно у нас массив состоящий из ссылок, так как ссылка это строка то и получается что массив стрингов.

vladimir-moshkov commented on Task_Animate_the_OOP_model/src/main/java/Hotel.java:
```diff
@@ -21,7 +22,7 @@ public void setListRooms(HotelRoom[] rooms) {
         for (int i = 0; i < listRooms.length; i++) {
```
Была какая-то договоренность использовать именно массивы, а не коллекции? 
С коллекциями намного удобней работать, и код становится более читабельный

scokpius replied:
Договоренности на использование массивов или коллекций не было и в условиях негде не звучало, согласна что с коллекциями удобнее работать.

vladimir-moshkov commented on Task_Animate_the_OOP_model/src/main/java/HotelSearch.java:
```diff
@@ -36,34 +36,35 @@ public void setDepartureDate(Data departureDate) {
-                   } else { if ((room.getGuests().length / 4) > 0 && (room.getGuests().length / 4) < 4 ){
-                                countNumberRooms = 1;
-                   }else {countNumberRooms = room.getGuests().length / 4;}}
+        if (room.getGuests().length > 0) {
```
Очень сложный нечитаемый код. Много вложенных ифов, которые сложно поддерживать.
подумай, как можно упростить, в том числе, используя различные библиотеки (например, Math)

vladimir-moshkov commented on Task_Animate_the_OOP_model/src/main/java/HotelSearch.java:
```diff
@@ -36,34 +36,35 @@ public void setDepartureDate(Data departureDate) {
+    public int countGuest() {
+        int countGuest = 0;
+        for (Room room : rooms) {
+            countGuest = room.getGuests().length;
```
что вернет этот метод, если будет три комнаты, в одной 1 гость ,в другой - 2, в третьей - 3 ?

vladimir-moshkov commented on Task_Animate_the_OOP_model/src/main/java/Price.java:
```diff
@@ -23,9 +23,9 @@ public void setSum(double sum) {
         String price;
-        if (this.currency.getSymbol() != ' '){
+        if (this.currency.getSymbol() != ' ') {
             price = String.format("%c %d", this.currency.getSymbol(), this.sum);
```
У тебя здесь идея не пишет никаких варнингов?

vladimir-moshkov reviewed (changes requested):
Нужна доработка

Commit:
Description of OOP Modal in code:
- entity classes created
- the attributes and methods necessary to work with them are described
- added getters and setters

Commit:
Refactoring the code:
- replacing arrays with lists
- coding code according to Code Convention

vladimir-moshkov commented on Task_Animate_the_OOP_model/src/main/java/Hotel.java:
```diff
@@ -0,0 +1,35 @@
+
+public class Hotel {
+    private String hotelName;
+    private List<HotelRoom>[] listRooms;
```
Стоп-стоп-стоп)
Что это за структура? Массив листов.
По идее ткут просто List должен быть

vladimir-moshkov commented on Task_Animate_the_OOP_model/src/main/java/Hotel.java:
```diff
@@ -0,0 +1,35 @@
+        return listRooms;
+    }
+
+    public void setListRooms(HotelRoom[] rooms) {
```
Должно быть так
public void setListRooms(List\<HotelRoom\> rooms)
А внутри используй addAll

vladimir-moshkov commented on Task_Animate_the_OOP_model/src/main/java/HotelRoom.java:
```diff
@@ -0,0 +1,63 @@
+        this.price = priceIn;
+        this.numberOfPeopleSleep = numberOfPeopleSleep;
+        this.areaRoom = areaRoom;
+        this.photoRoom = Arrays.copyOf(photoRoom, photoRoom.length);
```
никаких массивов. И используй addAll, 
Посмотри везде в похожих местах

vladimir-moshkov commented on Task_Animate_the_OOP_model/src/main/java/HotelSearch.java:
```diff
@@ -0,0 +1,65 @@
+
+    // На сайте так устроенно что в 1 комнату могут заселиться только 4 гостя разного возраста
+    // если гостей 5 то автомотически увелисивается количество комнот до 2
+    public int countNumberRooms(Room room) {
```
Писал замечание в прошлом ПР.
Абсолютно непонятно, для чего этот метод и почему на вход мы подаем Room,
Будешь править, напиши мне. Попробую объяснить

vladimir-moshkov commented on Task_Animate_the_OOP_model/src/main/java/Room.java:
```diff
@@ -0,0 +1,28 @@
+    }
+
+    public List<Guest> addGuest(Guest guest) {
+        for (int i = 0; i < guests.size(); i++) {
```
ЗАчем тут этот for? 
Делай просто guests.add(guest)

scokpius committed:
Refactoring the code:
- removal of incomprehensible constructions from the code of the List <String> [] type
- changing the code in terms of filling the collections

vladimir-moshkov commented on Task_Animate_the_OOP_model/src/main/java/HotelRoom.java:
```diff
@@ -0,0 +1,63 @@
+        this.price = priceIn;
+        this.numberOfPeopleSleep = numberOfPeopleSleep;
+        this.areaRoom = areaRoom;
+        this.photoRoom.addAll(photo);
```
Это конструктор, здесь ты инициализируешь свои переменные. 
addAll нужно использовать везде, где ты хочешь добавить к имеющейся коллекции новые значения. 
Написав так, у тебя будет NullPointerException. Разберись, почему, и как правильно инициализировать твою коллекцию.
Кстати, нужно бы подумать, действительно ли тебе и в сеттерах нужно addAll использовать

vladimir-moshkov commented on Task_Animate_the_OOP_model/src/main/java/HotelSearch.java:
```diff
@@ -0,0 +1,65 @@
+    // На сайте так устроенно что в 1 номер могут заселиться только 4 гостя разного возраста
+    // если гостей 5 то автомотически увелисивается количество номеров до 2
+    // функция нужна для определения количества номеров (может понадобиться для проверки)
+    public int countNumberRooms(List<Guest> guests) {  //расчитывает минимальный размер List<Room>
```
Созвонимся, расскажешь, для чего всё таки этот метод. Что-то я понять не могу

scokpius committed:
Refactoring the code:
- removing unnecessary setters
- initialization of collections, leaving the set method where it is necessary to override the collection
